pageserver/src/tenant.rs

```diff
@@ -1328,18 +1396,8 @@ impl Tenant {
         // and don't error out if some of the shutdown tasks have already been
         // completed!
 
-        // Stop the walreceiver first.
-        debug!("waiting for wal receiver to shutdown");
-        task_mgr::shutdown_tasks(
-            Some(TaskKind::WalReceiverManager),
-            Some(self.tenant_id),
-            Some(timeline_id),
-        )
-        .await;
-        debug!("wal receiver shutdown confirmed");
-
-        info!("waiting for timeline tasks to shutdown");
-        task_mgr::shutdown_tasks(None, Some(self.tenant_id), Some(timeline_id)).await;
+        debug!("waiting for timeline contexts to shutdown");
+        timeline.graceful_shutdown(false).await;
```

Comment:
As noted elsewhere, I think this PR should ensure that, after this call returns, no new tasks can operate on this timeline. E.g., by failing registration of request contexts for this timeline.